src/vs/base/common/paths.ts

```diff
@@ -392,3 +393,58 @@ export const isAbsoluteRegex = /^((\/|[a-zA-Z]:\\)[^\(\)<>\\'\"\[\]]+)/;
 export function isAbsolute(path: string): boolean {
 	return isAbsoluteRegex.test(path);
 }
+
+/**
+ * Shortens the paths but keeps them easy to distinguish.
+ * Replaces not important parts with ellipsis.
+ * Every shorten path matches only one original path and vice versa.
+ */
+export function shorten(paths: string[]): string[] {
+	var separator = isWindows ? '\\' : '/';
```

Comment:
Suggest to use `paths.nativeSep` here (https://github.com/Microsoft/vscode/blob/master/src/vs/base/common/paths.ts#L20)